README.md

```diff
@@ -117,42 +145,81 @@ Below are the few steps we need to do
       < JSON data removed for brevity. >
     }
     ```
-    Browse to your newly created web app and you should see a functioning web app. Replace <app_name> with a unique app name.
+Browse to your newly created web app, and you should see a functioning web app. Replace <appName> with the unique app name that you chose previously.
 
     ```
-    http://<app name>.azurewebsites.net
+    http://<appName>.azurewebsites.net
     ```
-    The above command also creates a Git-enabled app which allows you to deploy to azure from your local git. 
-    Local git is configured with url of 'https://<username>@<app_name>.scm.azurewebsites.net/<app_name>.git'
+The above command also creates a Git-enabled app which allows you to deploy to Azure from your local git. 
+Local Git repository is configured with this url: 'https://<username>@<AppName>.scm.azurewebsites.net/<AppName>.git'
+
+3. Deployment User
 
-- Create a deployment user
-    After the previous command is completed you can add add an Azure remote to your local Git repository. Replace <url> with the URL of the Git remote that you got from Enable Git for your app.
+After running the previous command, you can add an Azure Remote to your local Git repository. Replace <url> with the URL of the Git Remote that you got from [enabling Git for your app](https://docs.microsoft.com/en-us/azure/app-service/deploy-local-git).
 
     ```
     git remote add azure <url>
     ```
 
+### Configuring your Key Vault
+
+Use the [Azure Cloud Shell](https://shell.azure.com/bash) snippet below to create/get client secret credentials.
+
+- Create a service principal and configure its access to Azure resources:
+  ```Bash
+  az ad sp create-for-rbac -n <your-application-name> --skip-assignment
+  ```
+  Output:
+  ```json
+  {
+    "appId": "generated-app-ID",
+    "displayName": "your-application-name",
+    "name": "http://your-application-name",
+    "password": "random-password",
+    "tenant": "tenant-ID"
+  }
+  ```
+- Use the above returned credentials information to set **AZURE_CLIENT_ID**(appId), **AZURE_CLIENT_SECRET**(password) and **AZURE_TENANT_ID**(tenant) environment variables. The following example shows a way to do this in Bash:
+
+  ```Bash
+    export AZURE_CLIENT_ID="generated-app-ID"
+    export AZURE_CLIENT_SECRET="random-password"
+    export AZURE_TENANT_ID="tenant-ID"
+  ```
+
+- Grant the above mentioned application authorization to perform secret operations on the Key Vault:
+
+  ```Bash
+  az keyvault set-policy --name <your-key-vault-name> --spn $AZURE_CLIENT_ID --secret-permissions backup delete get list set
+  ```
+
+  > --secret-permissions:
+  > Accepted values: backup, delete, get, list, purge, recover, restore, set
+
+- Use the above mentioned Key Vault name to retrieve details of your Vault which also contains your Key Vault URL:
+  ```Bash
+  az keyvault show --name <your-key-vault-name>
```

Comment:
here you use lower+dashes, above you use Pascal and camel, please be consistent with all replacement tokens.  Please search this and all samples you are editing for this.